app/models/brand.rb

```diff
@@ -182,8 +184,47 @@ def in_trial?
     !!subscription.try(:trial?)
   end
 
+  def delete_account
+    ActiveRecord::Base.transaction do
+      unsubscribe_users_from_newsletter
+      subscription.cancel_plan! if subscription.present?
+      destroy
+    end
+  end
+
+  def unsubscribe_users_from_newsletter
+    users.each do |user|
+      user.update!(email_subscription: false) if user.email_subscription
+    end
+  end
+
+  def restore_brand
+    ActiveRecord::Base.transaction do
+      restore
+      restore_users
+      restore_subscription
+    end
+  end
+
   private
 
+  def restore_users
```

Comment:
From my above comment, I think moving away from restoring accounts is the way to go. It makes our lives easier.